src/spatialdata/_core/spatialdata.py

```diff
@@ -490,29 +490,72 @@ def transform_element_to_coordinate_system(
             The element to transform.
         target_coordinate_system
             The target coordinate system.
+        maintain_positioning
+            Default False (most common use case). If True, the data will be transformed but a transformation will be
+            added so that the positioning of the data in the target coordinate system will not change. If you want to
+            align datasets to a common coordinate system you should use the default value.
 
         Returns
         -------
         The transformed element.
         """
         from spatialdata import transform
-        from spatialdata.transformations import Identity
+        from spatialdata.transformations import Sequence
         from spatialdata.transformations.operations import (
+            get_transformation,
             get_transformation_between_coordinate_systems,
             remove_transformation,
             set_transformation,
         )
 
         t = get_transformation_between_coordinate_systems(self, element, target_coordinate_system)
-        transformed = transform(element, t, maintain_positioning=False)
-        remove_transformation(transformed, remove_all=True)
-        set_transformation(transformed, Identity(), target_coordinate_system)
-
+        if maintain_positioning:
+            transformed = transform(element, transformation=t, maintain_positioning=maintain_positioning)
+        else:
+            d = get_transformation(element, get_all=True)
+            assert isinstance(d, dict)
```

Comment:
Idem here, assert is not required I believe.
```suggestion
```

Reply:
makes mypy complain